src/term.rs

```diff
@@ -147,6 +146,25 @@ pub enum Term {
     ResolvedImport(FileId),
 }
 
+#[derive(Debug, Eq, PartialEq, Copy, Clone)]
+pub struct RecordAttrs {
+    pub open: bool,
+}
+
+impl Default for RecordAttrs {
+    fn default() -> Self {
+        RecordAttrs { open: false }
+    }
+}
+
+impl RecordAttrs {
+    pub fn merge(attrs1: RecordAttrs, attrs2: RecordAttrs) -> RecordAttrs {
+        RecordAttrs {
+            open: attrs1.open || attrs2.open,
+        }
+    }
+}
```

Comment:
(nit) This looks a lot like a monoid. Isn’t there a standard trait for that in Rust?

Reply:
Good point (and typical Haskeller :stuck_out_tongue: ), but I can't find one in the standard library, just in some third-party crates, so I guess I'll leave it at is it.